src/rpc/methods/f3.rs

```diff
@@ -541,6 +564,69 @@ impl RpcMethod<1> for F3GetF3PowerTable {
     }
 }
 
+/// F3Participate should be called by a storage provider to participate in signing F3 consensus.
+/// Calling this API gives the lotus node a lease to sign in F3 on behalf of given SP.
+/// The lease should be active only on one node. The lease will expire at the newLeaseExpiration.
+/// To continue participating in F3 with the given node, call F3Participate again before the newLeaseExpiration time.
+/// newLeaseExpiration cannot be further than 5 minutes in the future.
+/// It is recommended to call F3Participate every 60 seconds with newLeaseExpiration set 2min into the future.
+/// The oldLeaseExpiration has to be set to newLeaseExpiration of the last successful call.
+/// For the first call to F3Participate, set the oldLeaseExpiration to zero value/time in the past.
+/// F3Participate will return true if the lease was accepted. The minerID has to be the ID address of the miner.
+pub enum F3Participate {}
+impl RpcMethod<3> for F3Participate {
+    const NAME: &'static str = "Filecoin.F3Participate";
+    const PARAM_NAMES: [&'static str; 3] = [
+        "miner_address",
+        "new_lease_expiration",
+        "old_lease_expiration",
+    ];
+    const API_PATHS: ApiPaths = ApiPaths::V1;
+    const PERMISSION: Permission = Permission::Read;
```

Comment:
Given it modifies some internal node state, e.g., `F3_PARTICIPANT_LEASES`, shouldn't it be `write`?

Reply:
Good catch! The permission should be `Sign` according to the doc: https://github.com/filecoin-project/lotus/blob/master/documentation/en/api-v1-unstable-methods.md#f3participate